[clickhouse] Dependency update breaks build when cross-compiling

### Component(s)

exporter/clickhouse

### What happened?

Build fails when cross-compiling to ppc64le or win386.

```
cd ./cmd/otelcontribcol && GO111MODULE=on CGO_ENABLED=0 go build -trimpath -o ../../bin/otelcontribcol_linux_ppc64le \
	-ldflags "-X github.com/open-telemetry/opentelemetry-collector-contrib/internal/otelcontribcore/internal/version.Version=9d0020e" -tags "" .
# github.com/ClickHouse/clickhouse-go/v2/lib/column
Error: /home/runner/go/pkg/mod/github.com/!click!house/clickhouse-go/v2@v2.6.2/lib/column/fixed_string.go:100:40: too many arguments in call to binary.Str2Bytes
	have (string, int)
	want (string)
Error: /home/runner/go/pkg/mod/github.com/!click!house/clickhouse-go/v2@v2.6.2/lib/column/fixed_string.go:116:42: too many arguments in call to binary.Str2Bytes
	have (string, int)
	want (string)
Error: /home/runner/go/pkg/mod/github.com/!click!house/clickhouse-go/v2@v2.6.2/lib/column/fixed_string.go:142:31: too many arguments in call to binary.Str2Bytes
	have (string, int)
	want (string)
Error: /home/runner/go/pkg/mod/github.com/!click!house/clickhouse-go/v2@v2.6.2/lib/column/fixed_string.go:147:33: too many arguments in call to binary.Str2Bytes
	have (string, int)
	want (string)
Error: /home/runner/go/pkg/mod/github.com/!click!house/clickhouse-go/v2@v2.6.2/lib/column/string.go:74:50: too many arguments in call to binary.Str2Bytes
	have (string, int)
	want (string)
make: *** [Makefile:259: otelcontribcol] Error 2
Error: Process completed with exit code 2.
```

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4236028542/jobs/7360771718

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4235817053/jobs/7360591861


### Collector version

latest main

### Environment information

_No response_

### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

--- a/cmd/otelcontribcol/go.mod
+++ b/cmd/otelcontribcol/go.mod
@@ -213,7 +213,7 @@ require (
 	github.com/Azure/go-autorest/logger v0.2.1 // indirect
 	github.com/Azure/go-autorest/tracing v0.6.0 // indirect
 	github.com/ClickHouse/ch-go v0.52.1 // indirect
-	github.com/ClickHouse/clickhouse-go/v2 v2.6.2 // indirect
+	github.com/ClickHouse/clickhouse-go/v2 v2.6.1 // indirect
 	github.com/DataDog/agent-payload/v5 v5.0.73 // indirect
 	github.com/DataDog/datadog-agent/pkg/obfuscate v0.43.0-rc.3.0.20230206114529-17c7dfde736c // indirect
 	github.com/DataDog/datadog-agent/pkg/otlp/model v0.43.0-rc.3.0.20230206114529-17c7dfde736c // indirect

--- a/exporter/clickhouseexporter/go.mod
+++ b/exporter/clickhouseexporter/go.mod
@@ -3,7 +3,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/exporter/clickh
 go 1.19
 
 require (
-	github.com/ClickHouse/clickhouse-go/v2 v2.6.2
+	github.com/ClickHouse/clickhouse-go/v2 v2.6.1
 	github.com/cenkalti/backoff/v4 v4.2.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.0.0-00010101000000-000000000000
 	github.com/stretchr/testify v1.8.1

